# Assignment 3

### Instructions
1. Import the following modules:
    - `pandas`
    - `stats` from `scipy`
2. Create a dataframe from the data contained in *income_data.csv* using `pandas`

In [12]:
import pandas as pd
from scipy import stats
df = pd.read_csv("income_data.csv")

print(df)


                   job  type  income  education  prestige
0           accountant  prof      62         86        82
1                pilot  prof      72         76        83
2            architect  prof      75         92        90
3               author  prof      55         90        76
4              chemist  prof      64         86        90
5             minister  prof      21         84        87
6            professor  prof      64         93        93
7              dentist  prof      80        100        90
8             reporter    wc      67         87        52
9             engineer  prof      72         86        88
10          undertaker  prof      42         74        57
11              lawyer  prof      76         98        89
12           physician  prof      76         97        97
13      welfare.worker  prof      41         84        59
14             teacher  prof      48         91        73
15           conductor    wc      76         34        38
16          co

3. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *income* and *education*

In [13]:
r, p = stats.pearsonr(df['income'], df['education'])
print(f"Correlation: {r:.3f}")
print(f"P-Value: {p:.3f}")

Correlation: 0.725
P-Value: 0.000


3. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *income* and *prestige*

In [16]:
r, p = stats.pearsonr(df['income'], df['prestige'])
print(f"Correlation: {r:.3f}")
print(f"P-Value: {p:.3f}")

Correlation: 0.838
P-Value: 0.000


4. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *education* and *prestige*

In [18]:
r, p = stats.pearsonr(df['education'], df['prestige'])
print(f"Correlation: {r:.3f}")
print(f"P-Value: {p:.3f}")

Correlation: 0.852
P-Value: 0.000


5. In the empty `Markdown` cell below, explain the following for **each of the calculated correlation coefficients**
    - The **strength** and **direction** of the correlation
    - The **statistical significance** of the test (**p-value**); emphasize if we can trust the correlation test or not, and why

1. Income/Education
    - Strength/Direction: Strong and positive correlation (0.725)
    - Stat Sig: Statistically Significant (0.000...) = can trust this

2. Income/Prestige
    - Strength/Direction: Strong and positive correlation (0.838)
    - Stat Sig: Statistically Significant (0.000...) = can trust this

3. Education/Prestige
    - Strength/Direction: Strong and Positive Correlation
    - Stat Sig: Statistically Significant (0.000...) = can trust this

In the empty `Markdown` cell below, answer the following question:

6. Pearson's correlation coefficient OR covariance can both be used to measure the direction of the relationship between two variables (e.g., education and income). Explain why Pearson's correlation coefficient may be better than covariance.

Pearson’s correlation coefficient may be better because it standardizes the relationship between -1 and +1, making the strength and direction easier to understand and compare. Covariance depends on the units of the variables, which makes it harder to understand.

7. Submit this `Jupyter` file to D2L, renamed as **Last_First_Assignment3.ipynb** 
    - Replace '**Last**' and '**First**' with your name